File: ASP .NET/Clients/Dtos/ApiResponse.cs
using System.Text.Json.Serialization;

namespace Clients.Dtos;

/// <summary>
/// Respuesta genérica unificada para toda la API
/// Implementa un formato consistente: success, message, data, statusCode, timestamp
/// Equivalente a ApiResponse.java en la API JAVA
/// </summary>
public class ApiResponse<T>
{
    [JsonPropertyName("success")]
    public bool Success { get; set; }

    [JsonPropertyName("message")]
    public string? Message { get; set; }

    [JsonPropertyName("data")]
    public T? Data { get; set; }

    [JsonPropertyName("statusCode")]
    public int StatusCode { get; set; }

    [JsonPropertyName("timestamp")]
    public DateTime Timestamp { get; set; }

    /// <summary>
    /// Crea una respuesta exitosa
    /// </summary>
    public static ApiResponse<T> Ok(string message, T? data)
    {
        return new ApiResponse<T>
        {
            Success = true,
            Message = message,
            Data = data,
            StatusCode = 200,
            Timestamp = DateTime.UtcNow
        };
    }

    /// <summary>
    /// Crea una respuesta exitosa con solo mensaje
    /// </summary>
    public static ApiResponse<T> Ok(string message)
    {
        return Ok(message, default);
    }

    /// <summary>
    /// Crea una respuesta de error con status code
    /// </summary>
    public static ApiResponse<T> Error(int statusCode, string message)
    {
        return new ApiResponse<T>
        {
            Success = false,
            Message = message,
            Data = default,
            StatusCode = statusCode,
            Timestamp = DateTime.UtcNow
        };
    }

    /// <summary>
    /// Crea una respuesta de error 400
    /// </summary>
    public static ApiResponse<T> BadRequest(string message)
    {
        return Error(400, message);
    }

    /// <summary>
    /// Crea una respuesta de error 401
    /// </summary>
    public static ApiResponse<T> Unauthorized(string message)
    {
        return Error(401, message);
    }

    /// <summary>
    /// Crea una respuesta de error 403
    /// </summary>
    public static ApiResponse<T> Forbidden(string message)
    {
        return Error(403, message);
    }

    /// <summary>
    /// Crea una respuesta de error 404
    /// </summary>
    public static ApiResponse<T> NotFound(string message)
    {
        return Error(404, message);
    }

    /// <summary>
    /// Crea una respuesta de error 500
    /// </summary>
    public static ApiResponse<T> InternalServerError(string message)
    {
        return Error(500, message);
    }
}

File: JAVA/Clients/clients/src/main/java/com/futureprograms/clients/util/ApiResponse.java
package com.futureprograms.clients.util;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.http.HttpStatus;

import java.time.LocalDateTime;

/**
 * Respuesta genérica unificada para toda la API
 * Implementa un formato consistente: success, message, data, timestamp
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ApiResponse<T> {

    private Boolean success;
    private String message;
    private T data;
    private Integer statusCode;
    private LocalDateTime timestamp;

    /**
     * Crea una respuesta exitosa
     */
    public static <T> ApiResponse<T> ok(String message, T data) {
        return ApiResponse.<T>builder()
                .success(true)
                .message(message)
                .data(data)
                .statusCode(200)
                .timestamp(LocalDateTime.now())
                .build();
    }

    /**
     * Crea una respuesta de error
     */
    public static <T> ApiResponse<T> error(HttpStatus status, String message) {
        return ApiResponse.<T>builder()
                .success(false)
                .message(message)
                .statusCode(status.value())
                .timestamp(LocalDateTime.now())
                .build();
    }

    /**
     * Crea una respuesta de error con status code específico
     */
    public static <T> ApiResponse<T> error(int statusCode, String message) {
        return ApiResponse.<T>builder()
                .success(false)
                .message(message)
                .statusCode(statusCode)
                .timestamp(LocalDateTime.now())
                .build();
    }
}
